build/bp-core/bp-core-blocks.php

```diff
@@ -371,3 +394,129 @@ function bp_nouveau_is_primary_nav_widget_block_active() {
+	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classnames ) );
+
+	ob_start();
+
+	if ( $title ) :
+		?>
+		<h2 class="widget-title"><?php echo esc_html( $title ); ?></h2>
+		<?php
+	endif;
+
+	if ( is_user_logged_in() ) : ?>
+
+		<?php
+		/**
+		 * Fires before the display of widget content if logged in.
+		 *
+		 * @since 1.9.0
+		 */
+		do_action( 'bp_before_login_widget_loggedin' );	?>
+
+		<div class="bp-login-widget-user-avatar">
+			<a href="<?php echo bp_loggedin_user_domain(); ?>">
+				<?php bp_loggedin_user_avatar(
+					array(
+						'type'   => 'thumb',
+						'width'  => 50,
+						'height' => 50
+					)
+				); ?>
+			</a>
+		</div>
+
+		<div class="bp-login-widget-user-links">
+			<div class="bp-login-widget-user-link"><?php echo bp_core_get_userlink( bp_loggedin_user_id() ); ?></div>
+			<div class="bp-login-widget-user-logout"><a class="logout" href="<?php echo wp_logout_url( bp_get_requested_url() ); ?>"><?php _e( 'Log Out', 'buddypress' ); ?></a></div>
+		</div>
+
+		<?php
+
+		/**
+		 * Fires after the display of widget content if logged in.
+		 *
+		 * @since 1.9.0
+		 */
+		do_action( 'bp_after_login_widget_loggedin' ); ?>
+
+	<?php else : ?>
+
+		<?php
+
+		/**
+		 * Fires before the display of widget content if logged out.
+		 *
+		 * @since 1.9.0
+		 */
+		do_action( 'bp_before_login_widget_loggedout' ); ?>
+
+		<form name="bp-login-form" id="bp-login-widget-form" class="standard-form" action="<?php echo esc_url( site_url( 'wp-login.php', 'login_post' ) ); ?>" method="post">
+			<label for="bp-login-widget-user-login"><?php _e( 'Username', 'buddypress' ); ?></label>
+			<input type="text" name="log" id="bp-login-widget-user-login" class="input" value="" />
+
+			<label for="bp-login-widget-user-pass"><?php _e( 'Password', 'buddypress' ); ?></label>
+			<input type="password" name="pwd" id="bp-login-widget-user-pass" class="input" value="" <?php bp_form_field_attributes( 'password' ) ?> />
+
+			<div class="forgetmenot"><label for="bp-login-widget-rememberme"><input name="rememberme" type="checkbox" id="bp-login-widget-rememberme" value="forever" /> <?php _e( 'Remember Me', 'buddypress' ); ?></label></div>
+
+			<input type="submit" name="wp-submit" id="bp-login-widget-submit" value="<?php esc_attr_e( 'Log In', 'buddypress' ); ?>" />
+
+			<?php if ( bp_get_signup_allowed() ) : ?>
+
+				<span class="bp-login-widget-register-link"><a href="<?php echo esc_url( bp_get_signup_page() ); ?>"><?php _e( 'Register', 'buddypress' ); ?></a></span>
+
+			<?php endif; ?>
+
+			<?php
+
+			/**
+			 * Fires inside the display of the login widget form.
+			 *
+			 * @since 2.4.0
+			 */
+			do_action( 'bp_login_widget_form' ); ?>
+
+		</form>
+
+		<?php
+
+		/**
+		 * Fires after the display of widget content if logged out.
+		 *
+		 * @since 1.9.0
+		 */
+		do_action( 'bp_after_login_widget_loggedout' ); ?>
+
+	<?php endif;
+
+	$output = ob_get_clean();
+
```

Comment:
I tested the new block in post/page content, and it looked OK to me. I was guessing that those users who have added their own style rules that target the current widget would like for them to apply to the new block. Is that not how you're imagining it working?

Reply:
Sure. The above code avoids to add the wrapper into the Sidebar Widget and makes sure it is added into the post to apply the widget styles.

Sidebar widgets are using the wrapper defined by the theme (it could be `<section>` instead of `<div>`).